core/state/statedb.go

```diff
@@ -876,7 +876,11 @@ func (s *StateDB) IntermediateRoot(deleteEmptyObjects bool) common.Hash {
 	// Now we're about to start to write changes to the trie. The trie is so far
 	// _untouched_. We can check with the prefetcher, if it can give us a trie
 	// which has the same root, but also has some content loaded into it.
-	if prefetcher != nil {
+	//
+	// Don't check prefetcher if Verkle Trie has been used. In the context of Verkle,
+	// only a single trie is used for state hashing. Replacing a non-nil Verkle tree
+	// here could result in losing uncommitted changes from storage.
+	if prefetcher != nil && (s.trie == nil || !s.trie.IsVerkle()) {
```

Comment:
I am quite surprised that you need this, because in verkle mode the prefetcher should be `nil`. But in any case, it doesn't hurt to add that check.